frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,357 @@
+            G_reader_settings:saveSetting("hyph_user_dict", hyph_user_dict)
+            self:onChangedUserDictionary(true)
+        end,
+        hold_callback = function()
+            local hyph_user_dict = G_reader_settings:isTrue("hyph_user_dict")
+            UIManager:show(MultiConfirmBox:new{
+                text = hyph_user_dict and _("Would you like to enable or disable the user hyphenation dictionary?\n\nThe current default (★) is enabled.")
+                or _("Would you like to enable or disable the user hyphenation dictionary?\n\nThe current default (★) is disabled."),
+                choice1_text_func =  function()
+                    return hyph_user_dict and _("Disable") or _("Disable (★)")
+                end,
+                choice1_callback = function()
+                    G_reader_settings:makeFalse("hyph_user_dict")
+                end,
+                choice2_text_func = function()
+                    return hyph_user_dict and _("Enable (★)") or _("Enable")
+                end,
+                choice2_callback = function()
+                    G_reader_settings:makeTrue("hyph_user_dict")
+                end,
+            })
+        end,
+        checked_func = function()
+            return self:isAvailable()
+        end,
+        enabled_func = function()
+            return self:_enabled()
+        end,
+        separator = true,
+    }
+end
+
+-- add a button for dictquicklookup
+function ReaderUserHyph:addButton(buttons, pos, word, parentWidget)
+    if word:find("[ ,;-%.\n]") then return end
+    if not self:isAvailable() then return end
+
+    if G_reader_settings:readSetting("hyph_user_dict") then
+        table.insert(buttons, pos, {
+            {
+                id = "add_user_dict_entry",
+                text = T(_("Modify hyphenation of \"%1\"."), word),
+                callback = function()
+                    if parentWidget then
+                        UIManager:close(parentWidget)
+                    end
+                    self:modifyUserEntry(word)
+                end,
+            },
+        })
+    end
+end
+
+-------------------------------------------
+-- Helper functions for dictionary entries
+-------------------------------------------
+
+-- checks if suggestion is well formated
+function ReaderUserHyph:checkHyphenation(suggestion, word)
+    if suggestion:find("%-%-") then
+        return false -- two or more consecutive '-'
+    end
+
+    suggestion = suggestion:gsub("-","")
+    if self.ui.document:getLowercasedWord(suggestion) == self.ui.document:getLowercasedWord(word) then
+        return true -- characters match (case insensitive)
+    end
+    return false
+end
+
+-------------------------------------
+-- dictionary, file functions use:
+--    self.dict_file
+--    self.new_dict_file
+--    self.dict
+-------------------------------------
+
+-- opens user dictionary
+function ReaderUserHyph:openDictionary()
```

Comment:
Do this function and the next ones still need to be splitted, and have file handles stored in `self`, now that you no longer need to keep things opened across callback ?